File: backend/app/schemas/recommendations.py
from typing import Literal

from pydantic import BaseModel

# Categorias fixas de humor — cada uma mapeia pra um conjunto de gêneros
# TMDB (ver MOOD_GENRES em services/recommendations.py). Fechado por
# design: evita ter que lidar com humor livre/texto do usuário.
Mood = Literal["feliz", "triste", "emocionante", "assustador", "relaxante", "reflexivo"]


class RecommendationItem(BaseModel):
    tmdb_id: int
    media_type: str
    title: str
    overview: str
    poster_url: str | None
    release_date: str | None
    vote_average: float | None


class RecommendationsResponse(BaseModel):
    movies: list[RecommendationItem]
    shows: list[RecommendationItem]

File: frontend/src/lib/recommendations.ts
import { api } from "./api";
import type { MediaSummary } from "./media";

export type RecommendationsResponse = {
  movies: MediaSummary[];
  shows: MediaSummary[];
};

export type Mood = "feliz" | "triste" | "emocionante" | "assustador" | "relaxante" | "reflexivo";

export const MOODS: Mood[] = ["feliz", "triste", "emocionante", "assustador", "relaxante", "reflexivo"];

export async function getRecommendations(): Promise<RecommendationsResponse> {
  const { data } = await api.get<RecommendationsResponse>("/recommendations");
  return data;
}

export async function getMoodRecommendations(mood: Mood): Promise<RecommendationsResponse> {
  const { data } = await api.get<RecommendationsResponse>(`/recommendations/mood/${mood}`);
  return data;
}
